include/xtensor/xgrid_view.hpp

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay and Wolf Vollprecht          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_GRID_VIEW_HPP
+#define XTENSOR_GRID_VIEW_HPP
+
+#include "xstrided_view.hpp"
+#include "xnoalias.hpp"
+
+namespace xt
+{
+
+    template <class E>
+    class xchunk_generator
+    {
+    public:
+        template <class OE>
+        xchunk_generator(OE&& e, std::vector<std::size_t>& shape, std::vector<std::size_t>& chunk_shape);
+
+        xstrided_slice_vector& next();
+        bool last_chunk();
```

Comment:
This class should provide iterators over chunks (`chunk_begin()` / `chunk_end()`) instead of the couple `next()` / `last_chunk()`.